sesam.py

```diff
@@ -505,17 +509,64 @@ def __init__(self, args, logger):
         self.args = args
         self.logger = logger
         self.sesam_node = None
+        self.formatstyle = FormatStyle()
 
-    def read_config_file(self, filename):
-        parser = configparser.ConfigParser(strict=False)
-
+    def parse_config_file(self, filename):
+        config = {}
+        #try to parse as json, if fails parse as ini
         with open(filename) as fp:
-            parser.read_file(itertools.chain(['[sesam]'], fp), source=filename)
-            config = {}
-            for key, value in parser.items("sesam"):
-                config[key.lower()] = value
+            try:
+                config = json.load(fp)
+            except ValueError as e:
+                pass
+        if not config:
+            with open(filename) as fp:
+                parser = configparser.ConfigParser(strict=False)
+                #[sesam] section is prepended to support .syncconfig file
+                #  in which section is omitted
+                parser.read_file(itertools.chain(['[sesam]'], fp), source=filename)
+                config = {}
+                for section in parser.sections():
+                    for key, value in parser.items(section):
+                        config[key.lower()] = value
+
+        return config
+
+
+    def read_config_file(self, filename, is_required=False):
+        try:
```

Comment:
a code from _get_node_and_jwt_token_ is extracted as method here for reusability